peer_mentoring/groups/views.py

```diff
@@ -1,7 +1,61 @@
-from django.shortcuts import render
+from account_management.models import UserProfile
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import get_object_or_404, redirect, render
+from posts.models import Comment, Post
+
+from .forms import GroupPostCommentForm, GroupPostForm
+from .models import Group
 
 # Create your views here.
 
 
-def group_view(request):
-    pass
+@login_required
+def group_index_view(request):
+    groups = Group.objects.all()
+    context = {"groups": groups}
+    return render(request, "groups_index.html", context)
+
+
+@login_required
+def group_detail_view(request):
+    group = get_object_or_404(Group, pk=Group.id)
+    context = {
+        "group": group,
+    }
+    return render(request, "group_detail.html", context)
+
+
+@login_required
+def group_posts(request):
+    posts = Post.objects.all().filter(id=Group.id)
+    comments = request.POST.get(Comment)
+    return render(request, "group_detail.html", posts, comments)
+
+
+@login_required
+def group_members_index_view(request):
+    members = UserProfile.objects.all()
+    context = {"members": members}
+    return render(request, "group_detail.html", context)
+
+
+@login_required
+def group_create_post_view(request):
+    if request.method == "POST":
+        form = GroupPostForm(request.POST)
+        post = form.save()
+        return redirect("group-detail", post.id)
+    else:
+        form = GroupPostForm()
+    return render(request, "group_detail.html", {"form": form})
+
+
+@login_required
+def group_create_post_comment_view(request):
+    if request.method == "POST":
+        form = GroupPostCommentForm(request.POST)
+        comment = form.save()
```

Comment:
same: check if form if valid, if not you probably want to return the validated form object back so it shows the error(s) in the template

Reply:
Done